com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/ClassLoaderScanner.java

```diff
@@ -48,7 +48,9 @@ public ClassLoaderScanner(ClassLoader loader) {
     private static void scanDir(File dir, String packageName, Set<String> classes) {
         File[] files = dir.listFiles();
         for (File file : files) {
-            if (file.getName().endsWith(".class")) {
+            if (file.isDirectory()) {
+                scanDir(file, packageName + "." + file.getName(), classes);
+            } else if (file.getName().endsWith(".class")) {
```

Comment:
I made dir scanning recursive (I assume this will only be used in tests, since `java -jar bob.jar` usage has only a single jar file on the classpath).

Reply:
I think we use the `ClassLoaderScanner` to find and initialize the `Builders` and also implementors of the interfaces in `com.defold.extension.pipeline`?

Reply:
Yes, and it's only used in dev since both Bob and the editor only have jars on the classpath.